sklearn/_config.py

```diff
@@ -193,6 +195,16 @@ def set_config(
 
         .. versionadded:: 1.3
 
+    sparse_interface : str, default="spmatrix"
+
+        The sparse interface used for every sparse object that scikit-learn produces,
+        e.g., function returns, estimator attributes, estimator properties, etc.
+
+        - `"sparray"`: Return sparse as SciPy sparse array
+        - `"spmatrix"`: Return sparse as SciPy sparse matrix
```

Comment:
are we changing this default at some point? Should we introducing a deprecation cycle at the same time?